frontend/src/pages/signup/Verify.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react';
+import { CircularProgress, Typography } from '@material-ui/core';
+import {
+  AccountsService,
+  VerifyRegistration,
+  VerifyEmail,
+} from 'src/services/openapi';
+import { ContentHeader, ContentBox } from 'src/components';
+import { useSearchParams } from 'src/hooks';
+
+const executeVerifyUser = async (searchParams: Record<string, string>) => {
+  const { user_id, timestamp, signature } = searchParams;
+  const verificationData: VerifyRegistration = {
+    user_id,
+    timestamp: Number(timestamp),
+    signature,
+  };
+  return await AccountsService.accountsVerifyRegistrationCreate(
+    verificationData
+  );
+};
+
+const executeVerifyEmail = async (searchParams: Record<string, string>) => {
+  const { user_id, email, timestamp, signature } = searchParams;
+  const verificationData: VerifyEmail = {
+    user_id,
+    timestamp: Number(timestamp),
+    signature,
+    email,
+  };
+  return await AccountsService.accountsVerifyEmailCreate(verificationData);
+};
+
+const VerifySignature = ({ verify }: { verify: 'user' | 'email' }) => {
+  const searchParams = useSearchParams();
+  const [verificationState, setVerificationState] = useState<
+    'loading' | 'fail' | 'success'
+  >('loading');
+
+  useEffect(() => {
+    const verifyParams = async () => {
+      try {
+        if (verify === 'user') {
+          await executeVerifyUser(searchParams);
+        } else if (verify === 'email') {
+          await executeVerifyEmail(searchParams);
+        }
+        setVerificationState('success');
+      } catch (err) {
+        console.error(err);
+        setVerificationState('fail');
+      }
+    };
+    verifyParams();
```

Comment:
I never knew this trick after years of JavaScript 🤯

Reply:
If you refer to using an "async" function in a sync block, it's actually a common pattern in `useEffect()`.  
In general, that's not a silver bullet though: it works here as there is no return value, and the function catches all errors, to avoid unhandled promise rejections.